### ISSUE TITLE
[BUG] Arm exception when using love.graphics.clear on canvas after garbage collection of another canvas

### ISSUE THREAD

**Issue by Nehemek** [lovebrew/LovePotion#136]
### **Describe the bug**
I'm not sure how to describe it. To me it appears that love.graphics.clear is capable of overwriting used memory if it is called right after a garbage collection call. The code below is a sure way to reproduce the issue albeit maybe my description or interpretation may not be correct.

### **Expected behavior**
The garbage collector should safely dispose of the canvas space or a love.graphics.clear should be able to be performed safely after garbage collection.

### **Relevant Code To Reproduce**
```
function love.load()
    love.graphics.setBackgroundColor(0.8,0.75,1)
    canvas1 = love.graphics.newCanvas(200,240)
    love.graphics.setCanvas(canvas1)
    love.graphics.clear(1,1,1,1)
    love.graphics.setCanvas()
    canvas1 = nil
    collectgarbage("collect")
    canvas2 = love.graphics.newCanvas(200,240)
    love.graphics.setCanvas(canvas2)
    love.graphics.clear(1,1,1,1)
    love.graphics.setCanvas()
end

function love.gamepadpressed(joystick, button)
    if button == "start" then love.event.quit() end
end
```

### **[Screenshots]**
![armcrash](https://user-images.githubusercontent.com/66986051/116012775-38e2fa00-a5f2-11eb-83e1-f3fd778687fd.jpeg)

### **System Info [Please complete the following information]:**
System: 3ds v11.14.0-46U
CFW: Luma v10.2.1
Method of execution: Homebrew App
LÖVE Potion version: 2.1.0 (development version)

**Comment by TurtleP** [comment]
If I uh.. debugged those PC and LR addresses properly, it's breaking here:

```
0x00225e04: svcBreak at /home/fincs/pacman-packages/libctru/src/libctru-2.0.1/libctru/source/svc.s:401
0x00211e70: C3D_RenderTargetDelete at /home/fincs/pacman-packages/citro3d/src/citro3d-1.6.2/source/renderqueue.c:357
```

I'll see what I could do about this.

**Comment by TurtleP** [comment]
Fixed in 2fa4f8193753afcacad7e220930f81ec04ad16d6
